src/core/MemoryModel/Storage/Serialization/TaskDeserializer.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by rodion on 9/26/20.
+//
+
+#ifndef TODOLIST_SRC_CORE_MEMORYMODEL_STORAGE_SERIALIZATION_PROTO_TASKDESERIALIZER_H_
+#define TODOLIST_SRC_CORE_MEMORYMODEL_STORAGE_SERIALIZATION_PROTO_TASKDESERIALIZER_H_
+
+#include <src/core/MemoryModel/Task/TaskEntity.h>
+#include "task.pb.h"
+
+/*
+ * Class contains static methods to convert TaskProto to TaskEntity.
+ *
+ * @see task.proto
+ *
+ * @author Rodion Shyshkin
+ */
+
+class TaskDeserializer {
+ public:
+  static std::shared_ptr<TaskEntity>       DeserializeTask(const TaskProto&);
+
+ private:
+  /*
+   * \brief Converts TaskProto_Priority to Priority.
+   *
+   * @param TaskProrot_Priority
+   *
+   * @return Priority instance.
+   */
+  static Priority         DeserializePriority(const TaskProto_Priority&);
+};
+
```

Comment:
Shouldnt it be a virtual methods for mock in tests in future?

Reply:
No, because now it's just converter from TaskEntity into TaskProto.